src/components/RoadDamage/index.tsx

```diff
@@ -0,0 +1,261 @@
+        null,
+        newCancelToken()
+      )
+      if (autoWallet) {
+        const autoComputeJobs = await getComputeJobs(
+          chainIds,
+          autoWallet?.address,
+          null,
+          newCancelToken()
+        )
+        autoComputeJobs.computeJobs.forEach((job) => {
+          computeJobs.computeJobs.push(job)
+        })
+      }
+
+      setJobs(
+        // Filter computeJobs for dids configured in _constants
+        computeJobs.computeJobs.filter(
+          (job) =>
+            roadDamageAlgoDids.includes(job.algoDID) &&
+            job.status === 70 &&
+            job.results.filter((result) => result.filename === resultFileName)
+              .length > 0
+        )
+      )
+      setIsLoadingJobs(!computeJobs.isLoaded)
+    } catch (error) {
+      LoggerInstance.error(error.message)
+      setIsLoadingJobs(false)
+    }
+  }, [
+    chainIds,
+    roadDamageAlgoChains,
+    roadDamageAlgoDids,
+    accountId,
+    autoWallet,
+    newCancelToken
+  ])
+
+  useEffect(() => {
+    fetchJobs()
+  }, [refetchJobs, chainIds])
+
+  const addComputeResultToUseCaseDB = async (job: ComputeJobMetaData) => {
+    if (roadDamageList.find((row) => row.job.jobId === job.jobId)) {
+      toast.info('This compute job result already is part of the map view.')
+      return
+    }
+
+    const dataForSameInputExists =
+      roadDamageList.filter(
+        (row) =>
+          job.inputDID.filter((did) => row.job.inputDID.includes(did))
+            .length === job.inputDID.length
+      ).length > 0
+
+    if (dataForSameInputExists)
+      if (
+        !confirm(
+          'Compute job results for a job with the same dataset inputs already exists. Add anyways?'
+        )
+      )
+        return
+
+    try {
+      const datasetDDO = await getAsset(job.inputDID[0], newCancelToken())
+
+      const signerToUse =
+        job.owner.toLowerCase() === autoWallet?.address.toLowerCase()
+          ? autoWallet
+          : signer
+
+      const jobResult = await ProviderInstance.getComputeResultUrl(
+        datasetDDO.services[0].serviceEndpoint,
+        signerToUse,
+        job.jobId,
+        job.results.findIndex((result) => result.filename === resultFileName)
+      )
+
+      const binary = await getResultBinaryData(jobResult)
+      const resultData = await transformBinaryToRoadDamageResult(binary)
+
+      if (!resultData) return
+
+      const newuseCaseData: RoadDamageUseCaseData = {
+        job,
+        result: resultData
+      }
+
+      await createOrUpdateRoadDamage(newuseCaseData)
+      toast.success('Added a new compute result')
+    } catch (error) {
+      LoggerInstance.error(error)
+      toast.error('Could not add compute result')
+    }
+  }
+
+  const deleteJobResultFromDB = async (job: ComputeJobMetaData) => {
+    if (!confirm(`Are you sure you want to delete the result from map view?`))
+      return
+
+    const rowToDelete = roadDamageList.find(
+      (row) => row.job.jobId === job.jobId
```

Comment:
question: Can we be sure that job is never undefined?

Reply:
I added a statement on the `UseCaseProvider` ensuring rows always have jobs with IDs.
The `job` passed to the function will always be defined, as it is only used for valid entries in the `<ComputeJobs>` table component.